src/main/java/org/cactoos/CountOf.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos;
+
+import java.util.Iterator;
+import org.cactoos.func.UncheckedScalar;
+import org.cactoos.io.CountOfInput;
+import org.cactoos.list.CountOfIterable;
+import org.cactoos.list.CountOfIterator;
+import org.cactoos.text.TextOf;
+
+/**
+ * Length of iterator.
+ *
+ * <p>There is no thread-safety guarantee.
+ *
+ * @author Ix (ixmanuel@yahoo.com)
+ * @version $Id$
+ * @since 0.12
+ */
+public final class CountOf implements Scalar<Number> {
+
+    /**
+     * The underlying scalar number.
+     */
+    private final UncheckedScalar<?> scalar;
```

Comment:
@alex-semenyuk yes, it is an alternative but, I tried to make it clear and "ergonomic" for the end-user-developer. Alternatively, we could split counting for bytes and elements into: CountOf(Text, String, Iterable, Iterator) and SizeOf(Input, Bytes).